OneSignalDevelopers/ngosi
# Add formik logic and yup validation schema

Fixes: #69

## Solution
 I used yup to handle email validation by creating a validation schema called `sendMagicLinkSchema`.I then added formik logic to validate email input and display an error if there is any. Since the yup validation schema validates for inputs not click events, I created a new state called `yupError`. This stores errors that come from our onlick event triggered when clicking on the 'Send magic link' button and the error message gets displayed if there is any. I also added additional styling to compensate for the addition of formik elements (`<Formik>`,`<Form>` & `<Field>`).

## Testing

1. Start server; `yarn dev`
1. Start supabase ; `supabase start`
1. Make sure docker is running in the background
1. Navigate to the `/signin` page
1. Click on the 'Your Email' input element and unclick it by clicking on the 'Ngosi' header element. Observe that an error message will be displayed
1. Click on the 'Send magic link' button and notice that an error message will pop up

### Resources

- [Formik logic & yup validation scehma](https://formik.org/docs/guides/validation)
- [Formik with Typescript implementation](https://formik.org/docs/guides/typescript)
